validator/client/src/main/java/tech/pegasys/teku/validator/client/BeaconProposerPreparer.java

```diff
@@ -62,29 +62,32 @@ public void onSlot(UInt64 slot) {
     if (slot.mod(spec.getSlotsPerEpoch(slot)).isZero() || !firstCallDone) {
       firstCallDone = true;
 
-      SafeFuture<Optional<ProposerConfig>> proposerConfigFuture =
-          proposerConfigProvider.getProposerConfig();
-
-      validatorIndexProvider
-          .getValidatorIndexesByPublicKey()
-          .thenCompose(
-              publicKeyToIndex ->
-                  proposerConfigFuture
-                      .thenApply(
-                          proposerConfig ->
-                              buildBeaconPreparableProposerList(proposerConfig, publicKeyToIndex))
-                      .exceptionally(
-                          throwable -> {
-                            LOG.warn(
-                                "An error occurred while obtaining proposer config", throwable);
-                            return buildBeaconPreparableProposerList(
-                                Optional.empty(), publicKeyToIndex);
-                          }))
-          .thenAccept(validatorApiChannel::prepareBeaconProposer)
-          .finish(VALIDATOR_LOGGER::beaconProposerPreparationFailed);
+      calculateProposer();
     }
   }
 
+  private void calculateProposer() {
```

Comment:
nit: I think I'd call this `sendPreparableProposerList`.

Reply:
wow ok, that's not a name i'd have thought of but we'll go with that.